pyingest/tests/test_csgparsers.py

```diff
@@ -0,0 +1,81 @@
+"""
+Test parsers
+"""
+
+import unittest
+import filecmp
+import sys
+import os
+import glob
+import json
+
+from pyingest.parsers import oup
+from pyingest.parsers import iop
+from pyingest.config import config
+from pyingest.parsers.author_names import AuthorNames
+
+from pyingest.serializers import classic
+
+
+
+class TestOUP(unittest.TestCase):
+
+    def setUp(self):
+        stubdata_dir = os.path.join(os.path.dirname(__file__), 'data/stubdata/')
+        self.inputdocs = glob.glob(os.path.join(stubdata_dir, 'input/oup*'))
+        self.outputdir = os.path.join(stubdata_dir, 'parsed')
+        sys.stderr.write("test cases are: {}\n".format(self.inputdocs))
+
+    def test_oup_parser(self):
+        parser = oup.OUPJATSParser()
+        config.REFERENCE_TOPDIR = '/dev/null/'
+        for file in self.inputdocs:
+            # this will raise exceptions if something is wrong
+            with open(file, 'r') as fp:
+                test_data = parser.parse(fp)
+                self.assertIsNotNone(test_data, "%s: error reading doc" % file)
+            basefile = os.path.basename(file)
+            target = os.path.join(self.outputdir, basefile + '.json')
+            # save temporary copy of data structure
+            target_saved = target + '.json'
+            with open(target_saved, 'w') as fp:
+                json.dump(document, fp, sort_keys=True, indent=4)
```

Comment:
This is where I think the test is failing.  I know the parse method in oup creates a variable called document, but it doesn't survive outside of the call to parser.parse(fp) above.  Were you trying to access that, or something else here?